File: app/src/main/java/com/example/codex01/Actvities/LoginActivity.java
package com.example.codex01.Actvities;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import com.example.codex01.R;
import com.example.codex01.databinding.ActivityLoginBinding;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

public class LoginActivity extends BaseActivity {
    ActivityLoginBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityLoginBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        binding.Edemailaddress.setText("davesquare482@gmail.com");
        binding.Edpassword.setText("12345678");
        setVariable();
    }
    private void setVariable()
    {
        binding.loginBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String email = binding.Edemailaddress.getText().toString();
                String password = binding.Edpassword.getText().toString();
                if (!isValidEmail(email)) {
                    Toast.makeText(LoginActivity.this, "Please enter a valid email address", Toast.LENGTH_SHORT).show();
                    return;
                }
                if (password.length() < 6) {
                    Toast.makeText(LoginActivity.this, "Password must be at least 6 characters long", Toast.LENGTH_SHORT).show();
                    return;
                }
                if (!email.isEmpty() && !password.isEmpty()) {
                    DatabaseReference authRef = FirebaseDatabase.getInstance().getReference("Authentication");
                    authRef.orderByChild("EmailID").equalTo(email).addListenerForSingleValueEvent(new ValueEventListener() {
                        @Override
                        public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                            if (dataSnapshot.exists()) {
                                for (DataSnapshot snapshot : dataSnapshot.getChildren()) {
                                    String dbPassword = snapshot.child("Password").getValue(String.class);
                                    String name = snapshot.child("Name").getValue(String.class);
                                    String phonenumber = snapshot.child("Phone Number").getValue(String.class);
                                    if (dbPassword.equals(password)) {
                                        SharedPreferences sharedPreferences = getSharedPreferences("UserInfo", Context.MODE_PRIVATE);
                                        SharedPreferences.Editor editor = sharedPreferences.edit();
                                        editor.putString("name", name);
                                        editor.putString("email", email);
                                        editor.putString("password", password);
                                        editor.putString("phonenumber",phonenumber);
                                        editor.apply();
                                        Toast.makeText(LoginActivity.this, "Login Successful", Toast.LENGTH_SHORT).show();
                                        startActivity(new Intent(LoginActivity.this, MainActivity.class));
                                        return;
                                    }
                                }
                                Toast.makeText(LoginActivity.this, "Password incorrect", Toast.LENGTH_SHORT).show();
                            } else {
                                Toast.makeText(LoginActivity.this, "Account doesn't exist, please SignUp", Toast.LENGTH_SHORT).show();
                            }
                        }

                        @Override
                        public void onCancelled(@NonNull DatabaseError databaseError) {
                            // Handle Errors
                        }
                    });
                } else {
                    Toast.makeText(LoginActivity.this, "Please Enter the Email and Password", Toast.LENGTH_SHORT).show();
                }
            }
        });
        binding.linkSignUp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(LoginActivity.this,SignUpActivity.class));
            }
        });
    }
    private boolean isValidEmail(CharSequence target) {
        return !TextUtils.isEmpty(target) && android.util.Patterns.EMAIL_ADDRESS.matcher(target).matches();
    }
}

File: MyApplication/app/src/main/java/com/example/myapplication/Activities/LoginActivity.kt
package com.example.myapplication.Activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.util.Patterns
import android.widget.Toast
import com.example.myapplication.databinding.ActivityLoginBinding
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener


class LoginActivity : BaseActivity() {
    var binding: ActivityLoginBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLoginBinding.inflate(layoutInflater)
        setContentView(binding!!.root)
        binding?.Edemailaddress?.setText("davesquare482@gmail.com")
        binding?.Edpassword?.setText("12345678")
        setVariable()
    }

    private fun setVariable() {
        binding!!.loginBtn.setOnClickListener { view ->
            val email = binding!!.Edemailaddress.text.toString()
            val password = binding!!.Edpassword.text.toString()
            if (!isValidEmail(email)) {
                Toast.makeText(
                    this@LoginActivity,
                    "Please enter a valid email address",
                    Toast.LENGTH_SHORT
                ).show()
                return@setOnClickListener
            }
            if (password.length < 6) {
                Toast.makeText(
                    this@LoginActivity,
                    "Password must be at least 6 characters long",
                    Toast.LENGTH_SHORT
                ).show()
                return@setOnClickListener
            }
            if (!email.isEmpty() && !password.isEmpty()) {
                val authRef = FirebaseDatabase.getInstance().getReference("Authentication")
                authRef.orderByChild("EmailID").equalTo(email)
                    .addListenerForSingleValueEvent(object : ValueEventListener {
                        override fun onDataChange(dataSnapshot: DataSnapshot) {
                            if (dataSnapshot.exists()) {
                                for (snapshot in dataSnapshot.children) {
                                    val dbPassword = snapshot.child("Password").getValue(String::class.java)
                                    val name = snapshot.child("Name").getValue(String::class.java)
                                    val phoneNumber = snapshot.child("Phone Number").getValue(String::class.java)
                                    if (dbPassword == password) {
                                        val sharedPreferences = getSharedPreferences(
                                            "UserInfo",
                                            Context.MODE_PRIVATE
                                        )
                                        val editor = sharedPreferences.edit()
                                        editor.putString("name", name)
                                        editor.putString("email", email)
                                        editor.putString("password", password)
                                        editor.putString("phonenumber", phoneNumber)
                                        editor.apply()
                                        Toast.makeText(
                                            this@LoginActivity,
                                            "Login Successful",
                                            Toast.LENGTH_SHORT
                                        ).show()
                                        startActivity(Intent(this@LoginActivity, MainActivity::class.java))
                                        return
                                    }
                                }
                                Toast.makeText(
                                    this@LoginActivity,
                                    "Password incorrect",
                                    Toast.LENGTH_SHORT
                                ).show()
                            } else {
                                Toast.makeText(
                                    this@LoginActivity,
                                    "Account doesn't exist, please SignUp",
                                    Toast.LENGTH_SHORT
                                ).show()
                            }
                        }

                        override fun onCancelled(databaseError: DatabaseError) {
                            // Handle Errors
                        }
                    })
            } else {
                Toast.makeText(
                    this@LoginActivity,
                    "Please Enter the Email and Password",
                    Toast.LENGTH_SHORT
                ).show()
            }
        }

        binding!!.linkSignUp.setOnClickListener {
            startActivity(
                Intent(
                    this@LoginActivity,
                    SignUpActivity::class.java
                )
            )
        }
    }
    private fun isValidEmail(target: CharSequence): Boolean {
        return !TextUtils.isEmpty(target) && Patterns.EMAIL_ADDRESS.matcher(target).matches()
    }
}
